components/camel-cxf/src/main/java/org/apache/camel/component/cxf/jaxrs/CxfRsProducer.java

```diff
@@ -865,6 +865,7 @@ public JAXRSClientFactoryBean get(String address) throws Exception {
                     LOG.trace("Created client factory bean and add to cache for address '{}'", address);
                     
                 } else {
+                    retVal.setAddress(address);
```

Comment:
I don't think that the `ClientFactoryBeanCache` would be responsible to mutate its content. However, I've noticed that the `ClientFactoryBeanCache` contains multiple soft references to the same underlying object and I wonder if it could be the cause of the issue.